lib/modules/datasource/docker/index.ts

```diff
@@ -883,20 +891,41 @@ export class DockerDatasource extends Datasource {
       packageName,
       registryUrl!
     );
-    let tags: string[] | null = null;
-    if (registryHost === 'https://index.docker.io') {
-      tags = await this.getDockerHubTags(dockerRepository);
-    }
-    tags ??= await this.getTags(registryHost, dockerRepository);
-    if (!tags) {
+
+    type TagsResultType = AsyncResult<
+      Release[],
+      NonNullable<Error | 'tags-fetch-error'>
+    >;
+
+    const getTags = (): TagsResultType =>
+      Result.wrapNullable(
+        this.getTags(registryHost, dockerRepository),
+        'tags-fetch-error' as const
+      ).transform((tags) => tags.map((version) => ({ version })));
+
+    const getDockerHubTags = (): TagsResultType => {
+      const dockerHubCache = new DockerHubCache(this.http, dockerRepository);
+      return Result.wrap(dockerHubCache.getTags()).catch(getTags);
+    };
+
+    const tagsResult =
+      registryHost === 'https://index.docker.io' &&
+      process.env.RENOVATE_X_DOCKER_HUB_TAGS
+        ? getDockerHubTags()
+        : getTags();
```

Comment:
Does this still fall back to the old approach (index.docker.io) if fetching from hub.docker.com fails?

Reply:
Yes, I tried to stick to current logic as much as possible